src/Common/VersionNumber.h

```diff
@@ -0,0 +1,52 @@
+#include <tuple>
+#include <string>
+#include <vector>
+#include <iostream>
+
+namespace DB
+{
+
+/// Simple numeric version representation.
+///
+/// Supports only "major.minor.patch", other components are ignored.
+struct VersionNumber
```

Comment:
I don't like this struct, let's move it completely to SettingsQuirks.cpp

Reply:
Do you dislike something specific?
Initially I placed it in SettingsQuirks, but later decided that it is common enough and that I want to cover it with unit test.
Plus there is another such place - renameat2 detection, that may use the same code.